yjit/src/backend/x86_64/mod.rs

```diff
@@ -139,6 +139,32 @@ impl Assembler
             }
         }
 
+        fn split_comparison_opnds(asm: &mut Assembler, unmapped_opnds: &Vec<Opnd>, left: &Opnd, right: &Opnd) -> (Opnd, Opnd) {
+            let split_opnd = match (unmapped_opnds[0], unmapped_opnds[1]) {
```

Comment:
Detail here but, do you really need `unmapped_opnds` ? Couldn't you just pattern match with `left` and `right` directly?

Reply:
+1. Could `split_arithmetic_opnds` work that way as well?

Reply:
CC @kddnewton

Reply:
You don't need `unmapped_opnds` here. I needed them for `split_arithmetic_opnds` because you need to compare the previous live ranges. But you can just check `left` and `right` here. And actually we should just inline this whole method since they're just getting assigned anyway.

Reply:
Having a separate method can be good for readability but we should remove the unused params

Reply:
Hey Kevin. Can you remove the unused params and we can merge this PR?

Reply:
@maximecb actually at this point I think this PR has been subsumed by a couple of other PRs and pairing sessions. Sorry I forgot this one was open. At the moment the current `Insn::Cmp` and `Insn::Test` are doing this optimization.